grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# Lint as: python3
+import os
+import json
+import typing
+from typing import Any, List, Tuple, Dict, Text
+
+from werkzeug.wrappers import Request, Response
+from werkzeug.wrappers.json import JSONMixin
+from werkzeug.routing import Map, Rule
+from werkzeug.exceptions import HTTPException, NotFound
+from werkzeug.middleware.shared_data import SharedDataMiddleware
+from werkzeug.utils import redirect
+from werkzeug.serving import run_simple
+
+from absl import app
+from absl import flags
+
+from grr_response_core.config import server as config_server
+from grr_response_server import fleetspeak_connector
+from grr_response_server import fleetspeak_utils
+from grr_response_server import server_startup
+from grr_response_core import config
+from grr_response_core.config import contexts
+
+from fleetspeak.src.server.proto.fleetspeak_server.resource_pb2 import ClientResourceUsageRecord
+
+
+AVAILABLE_METRICS = ['mean_user_cpu_rate', 'max_user_cpu_rate', 'mean_system_cpu_rate', 
+                    'max_system_cpu_rate', 'mean_resident_memory_mib', 'max_resident_memory_mib']
+
+flags.DEFINE_bool(
+  "version",
+  default=False,
+  allow_override=True,
+  help="Print the GRR console version number and exit immediately.")
+
+
+class JSONRequest(JSONMixin, Request):
+  pass
+
+
+class JSONResponse(JSONMixin, Response):
+  pass
+
+
+class Grrafana(object):
+  """GRRafana HTTP server instance.
+  
+  A full description of all endpoints implemented within this HTTP
+  server can be found in https://github.com/simPod/grafana-json-datasource#api."""
+
+  def __init__(self, config: dict) -> None:
+    """Constructor."""
+    self.url_map = Map([
+    Rule('/', endpoint='Root', methods=["GET"]),
+    Rule('/search', endpoint='Search', methods=["POST"]),
+    Rule('/query', endpoint='Query', methods=["POST"]),
+    Rule('/annotations', endpoint='Annotations', methods=["POST"])
+    ])
+
+  def dispatchRequest(self, request: JSONRequest) -> Any:
```

Comment:
We use the PascalCase (a variation of camelCase with first character capitalized) for class, methods and function names throughout the code. This should be `DispatchRequest`.